activerecord/lib/active_record/associations/builder/has_and_belongs_to_many.rb

```diff
@@ -74,8 +74,8 @@ def self.add_right_association(name, options)
           self.right_reflection = _reflect_on_association(rhs_name)
         end
 
-        def self.retrieve_connection
-          left_model.retrieve_connection
+        def self.retrieve_connection(spec_name = nil)
```

Comment:
I see no problem with that - happy to chnage.  For my own edification, why do you prefer that?